Feature: Show size in byte

Lzzzzzt opened the pull request:
Add a cli option that show file size in bytes rather than KB or MB, etc.
solve  the #1236.

svenstaro commented:
Sorry for ignoring this for so long! Would you perhaps like to rebase this? I promise I'll review it quickly. :)

Lzzzzzt commented:
> Sorry for ignoring this for so long! Would you perhaps like to rebase this? I promise I'll review it quickly. :)

OK

Lzzzzzt commented:
> Sorry for ignoring this for so long! Would you perhaps like to rebase this? I promise I'll review it quickly. :)

I'm done

svenstaro commented:
Ok, so I'm thinking: Perhaps this should be a frontend option like the sorting method and sorting order? It would be consistent since it's a display thing like those other options. There could be a button in the "Size" row that would allow you to toggle this (and it'd change the query parameter).

We can still have something like `--default-size-display` which is either `human` or `exact`. What do you think about that?

Lzzzzzt commented:
> Ok, so I'm thinking: Perhaps this should be a frontend option like the sorting method and sorting order? It would be consistent since it's a display thing like those other options. There could be a button in the "Size" row that would allow you to toggle this (and it'd change the query parameter).
> 
> We can still have something like `--default-size-display` which is either `human` or `exact`. What do you think about that?

Agreed! Keeping it consistent with sorting options and adding a toggle in the “Size” row makes sense. I’m good with the --default-size-display flag too.

Lzzzzzt commented:
> There could be a button in the "Size" row that would allow you to toggle this (and it'd change the query parameter).

It seems there is no suitable place for the button in the table header, I think the `Size` string in the entry could be a button to change the display behavior. And this will change other entries' display behavior, not only the clicked one.

svenstaro commented:
Do you want to rebase this and for the time being just exchange `--show_size_in_byte` for `--size-display` with an `enum` that is either `human` or `exact`? I'd like to get this into the next release if possible. Let's ignore the frontend stuff for the time being.

Lzzzzzt commented:
> Do you want to rebase this and for the time being just exchange `--show_size_in_byte` for `--size-display` with an `enum` that is either `human` or `exact`? I'd like to get this into the next release if possible. Let's ignore the frontend stuff for the time being.

Ok

Lzzzzzt commented:
> Do you want to rebase this and for the time being just exchange `--show_size_in_byte` for `--size-display` with an `enum` that is either `human` or `exact`? I'd like to get this into the next release if possible. Let's ignore the frontend stuff for the time being.

I've rebased and changed cli args, could you review the code?

svenstaro commented on src/renderer.rs:
```diff
@@ -554,13 +555,19 @@ fn entry_row(
-                                    (build_link("size", &format!("{}", size), sort_method, sort_order))
+                                @if show_exact_bytes {
+                                    span.mobile-info.size {
+                                        (maud::display(format!("{}B", size.as_u64())))
```
Shall we display it the same as the human format with `{} B` instead of `{}B`?
